server/src/main/java/io/crate/execution/engine/aggregation/impl/average/IntervalAverageAggregation.java

```diff
@@ -0,0 +1,283 @@
+            return Objects.equals(that.value(), value());
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(value());
+        }
+    }
+
+    public static class IntervalAverageStateType extends DataType<IntervalAverageState>
+        implements FixedWidthType, Streamer<IntervalAverageState> {
+
+        public static final int ID = 1028;
+        private static final IntervalAverageStateType INSTANCE = new IntervalAverageStateType();
+        private static final int AVERAGE_STATE_SIZE = (int) RamUsageEstimator.shallowSizeOfInstance(IntervalAverageState.class);
+
+        @Override
+        public int id() {
+            return ID;
+        }
+
+        @Override
+        public Precedence precedence() {
+            return Precedence.CUSTOM;
+        }
+
+        @Override
+        public String getName() {
+            return "interval_average_state";
+        }
+
+        @Override
+        public Streamer<IntervalAverageState> streamer() {
+            return this;
+        }
+
+        @Override
+        public IntervalAverageState sanitizeValue(Object value) {
+            return (IntervalAverageState) value;
+        }
+
+        @Override
+        public int compare(IntervalAverageState val1, IntervalAverageState val2) {
+            if (val1 == null) return -1;
+            return val1.compareTo(val2);
+        }
+
+        @Override
+        public IntervalAverageState readValueFrom(StreamInput in) throws IOException {
+            IntervalAverageState averageState = new IntervalAverageState();
+            averageState.sum = new Period(in.readVLong());
+            averageState.count = in.readVLong();
+            return averageState;
+        }
+
+        @Override
+        public void writeValueTo(StreamOutput out, IntervalAverageState v) throws IOException {
+            out.writeVLong(v.sum.toStandardDuration().getMillis());
+            out.writeVLong(v.count);
+        }
+
+        @Override
+        public int fixedSize() {
+            return AVERAGE_STATE_SIZE;
+        }
+
+        @Override
+        public long valueBytes(IntervalAverageState value) {
+            return fixedSize();
+        }
+    }
+
+    private final Signature signature;
+    private final BoundSignature boundSignature;
+
+    IntervalAverageAggregation(Signature signature, BoundSignature boundSignature) {
+        this.signature = signature;
+        this.boundSignature = boundSignature;
+    }
+
+    @Override
+    public IntervalAverageState iterate(RamAccounting ramAccounting,
+                                        MemoryManager memoryManager,
+                                        IntervalAverageState state,
+                                        Input<?>... args) {
+        if (state != null) {
+            Period value = (Period) args[0].value();
+            if (value != null) {
+                state.addPeriod(value); // Mutates state.
```

Comment:
Can we add a test that avg of 2 nulls is null? It will also cover ` if (count > 0) { ... else return null`